src/tpm2_wrap.c

```diff
@@ -1027,6 +1061,62 @@ int wolfTPM2_CreateAndLoadKey(WOLFTPM2_DEV* dev, WOLFTPM2_KEY* key,
     return rc;
 }
 
+int wolfTPM2_CreateLoadedKey(WOLFTPM2_DEV* dev, WOLFTPM2_KEYBLOB* keyBlob,
+    WOLFTPM2_HANDLE* parent, TPMT_PUBLIC* publicTemplate,
+    const byte* auth, int authSz)
+{
+    int rc, authCount = 0;
+    CreateLoaded_In  createLoadedIn;
+    CreateLoaded_Out createLoadedOut;
+
+    if (dev == NULL || keyBlob == NULL || parent == NULL || publicTemplate == NULL)
+        return BAD_FUNC_ARG;
+
+    /* clear output key buffer */
+    XMEMSET(keyBlob, 0, sizeof(WOLFTPM2_KEYBLOB));
+    XMEMSET(&createLoadedOut, 0, sizeof(createLoadedOut)); /* make sure pub struct is zero init */
+
+    if (!parent->policyAuth) {
+        /* Set password auth after any pre-existing sessions, e.g. Policy */
+        authCount = TPM2_GetSessionAuthCount(&dev->ctx);
+        /* set session auth for parent key */
+        wolfTPM2_SetAuthHandle(dev, authCount-1, parent);
+    }
+
+    XMEMSET(&createLoadedIn, 0, sizeof(createLoadedIn));
+    createLoadedIn.parentHandle = parent->hndl;
+    if (auth) {
+        createLoadedIn.inSensitive.sensitive.userAuth.size = authSz;
+        XMEMCPY(createLoadedIn.inSensitive.sensitive.userAuth.buffer, auth,
+            createLoadedIn.inSensitive.sensitive.userAuth.size);
+    }
+    XMEMCPY(&createLoadedIn.inPublic.publicArea, publicTemplate, sizeof(TPMT_PUBLIC));
+
+    rc = TPM2_CreateLoaded(&createLoadedIn, &createLoadedOut);
+    if (rc != TPM_RC_SUCCESS) {
+    #ifdef DEBUG_WOLFTPM
+        printf("TPM2_CreateLoaded key failed %d: %s\n", rc, wolfTPM2_GetRCString(rc));
+    #endif
+        return rc;
+    }
+
+#ifdef DEBUG_WOLFTPM
+    printf("TPM2_CreateLoaded key: pub %d, priv %d\n",
+        createLoadedOut.outPublic.size, createLoadedOut.outPrivate.size);
+    TPM2_PrintPublicArea(&createLoadedOut.outPublic);
+#endif
+
+    keyBlob->handle.auth = createLoadedIn.inSensitive.sensitive.userAuth;
+    keyBlob->handle.symmetric = createLoadedOut.outPublic.publicArea.parameters.asymDetail.symmetric;
+
+    keyBlob->handle.hndl = createLoadedOut.objectHandle;
+    keyBlob->pub = createLoadedOut.outPublic;
```

Comment:
Please use the new `wolfTPM2_CopyPub`, `wolfTPM2_CopyPriv` and `wolfTPM2_CopyName`